flytekit/models/literals.py

```diff
@@ -574,7 +575,7 @@ def from_flyte_idl(cls, pb2_object: _literals_pb2.StructuredDatasetMetadata) ->
 
 
 class StructuredDataset(_common.FlyteIdlEntity):
-    def __init__(self, uri: str, metadata: StructuredDatasetMetadata):
+    def __init__(self, uri: str, metadata: typing.Optional[StructuredDatasetMetadata] = None):
```

Comment:
same here - if metadata is made optional, then to_flyte_idl and from_flyte_idl will have to change.  Otherwise, you'll end up calling:

self.metadata.to_flyte_idl(), which will resolve to `None.to_flyte_idl()` which will obviously fail.